File: modules/java_workloads/mem/MemStringChurn.java
// One-shot string-churn workload: massive transient allocation traffic
// with a tiny live heap at dump time.

package monorepo.workloads;

public final class MemStringChurn {
    private static final int DEFAULT_PIECES = 8000;

    private MemStringChurn() {
    }

    public static void main(String[] args) throws Exception {
        int pieces = WorkloadN.workloadN(DEFAULT_PIECES);
        ProfDump.start();
        String built = StringChurn.concat(pieces, "0123456789abcdef");
        String out = ProfDump.dump();
        System.out.printf("built %d chars; heap recording: %s%n", built.length(), out);
    }
}

File: modules/python_workloads/mem/mem_string_churn.py
"""One-shot string-churn workload: massive transient allocation traffic
with a tiny live heap at dump time."""

from prof_dump import run_profiled
from python_workloads.string_churn import concat
from python_workloads.workload_n import workload_n

DEFAULT_PIECES = 8000


def main() -> None:
    pieces = workload_n(DEFAULT_PIECES)
    s, out = run_profiled(lambda: concat(pieces, "0123456789abcdef"))
    print(f"built {len(s)} chars; heap profile: {out}")


if __name__ == "__main__":
    main()
